reposerver/cache/cache.go

```diff
@@ -69,6 +74,43 @@ func (c *Cache) SetApps(repoUrl, revision string, apps map[string]string) error
 	return c.cache.SetItem(listApps(repoUrl, revision), apps, c.repoCacheExpiration, apps == nil)
 }
 
+func helmIndexRefsKey(repo string) string {
+	return fmt.Sprintf("helm-index|%s", repo)
+}
+
+func (c *Cache) SetHelmIndex(repo string, indexData []byte) error {
+	return c.cache.SetItem(helmIndexRefsKey(repo), indexData, c.revisionCacheExpiration, false)
+}
+
+func (c *Cache) GetHelmIndex(repo string, indexData *[]byte) error {
+	return c.cache.GetItem(helmIndexRefsKey(repo), indexData)
+}
+
+func gitRefsKey(repo string) string {
+	return fmt.Sprintf("git-refs|%s", repo)
+}
+
+func (c *Cache) SetGitReferences(repo string, references []*plumbing.Reference) error {
+	var input [][2]string
+	for i := range references {
+		input = append(input, references[i].Strings())
+	}
+	return c.cache.SetItem(gitRefsKey(repo), input, c.revisionCacheExpiration, false)
+}
+
+func (c *Cache) GetGitReferences(repo string, references *[]*plumbing.Reference) error {
+	var input [][2]string
+	if err := c.cache.GetItem(gitRefsKey(repo), &input); err != nil {
+		return err
+	}
+	var res []*plumbing.Reference
+	for i := range input {
+		res = append(res, plumbing.NewReferenceFromStrings(input[i][0], input[i][1]))
+	}
+	*references = res
+	return nil
```

Comment:
Can you please add some comments to the public methods?

Reply:
done